s2/loop.go

```diff
@@ -1287,11 +1291,36 @@ func (l *Loop) decode(d *decoder) {
 		return
 	}
 	l.vertices = make([]Point, nvertices)
-	for i := range l.vertices {
-		l.vertices[i].X = d.readFloat64()
-		l.vertices[i].Y = d.readFloat64()
-		l.vertices[i].Z = d.readFloat64()
+	floatSize := 8
+	vertexSize := 3 * floatSize
```

Comment:
declare global constants for floatSize (which definitely needs a better name) and vertexSize.